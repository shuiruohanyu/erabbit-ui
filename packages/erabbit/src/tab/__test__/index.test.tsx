import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import { ref } from 'vue'
import { Tab, TabPanel } from '..'
import { later } from '../../test'

test('should render default Tab', async () => {
  const wrapper = mount({
    setup() {
      return () => (
        <Tab>
          <TabPanel label="首页" name="first">
            1
          </TabPanel>
          <TabPanel label="用户管理" name="second">
            2
          </TabPanel>
          <TabPanel label="角色管理" name="third">
            3
          </TabPanel>
        </Tab>
      )
    },
  })
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render small size Tab', async () => {
  const wrapper = mount(
    <Tab size="small">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render  mini size Tab', async () => {
  const wrapper = mount(
    <Tab size="mini">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.mini').exists()).toBe(true)
})
test('should render card type Tab', async () => {
  const wrapper = mount(
    <Tab type="card">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.card').exists()).toBe(true)
})
test('should render border-card type Tab', async () => {
  const wrapper = mount(
    <Tab type="border-card">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.border-card').exists()).toBe(true)
})

test('should render left position Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="left">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.left').exists()).toBe(true)
})
test('should render right position Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="right">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render bottom position Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="bottom">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.bottom').exists()).toBe(true)
})
test('should render bottom position and card type Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="bottom" type="card">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.bottom').exists()).toBe(true)
  expect(wrapper.find('.er-tabs.card').exists()).toBe(true)
})
test('should render bottom position and border-card type and mini size Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="bottom" type="border-card" size="mini">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.bottom').exists()).toBe(true)
  expect(wrapper.find('.er-tabs.border-card').exists()).toBe(true)
})
test('should render bottom position and small size Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="bottom" size="small">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.bottom').exists()).toBe(true)
  expect(wrapper.find('.er-tabs.small').exists()).toBe(true)
})
test('should render bottom position and mini size Tab', async () => {
  const wrapper = mount(
    <Tab tabPosition="bottom" size="mini">
      <TabPanel label="首页" name="first">
        1
      </TabPanel>
      <TabPanel label="用户管理" name="second">
        2
      </TabPanel>
      <TabPanel label="角色管理" name="third">
        3
      </TabPanel>
    </Tab>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs.bottom').exists()).toBe(true)
  expect(wrapper.find('.er-tabs.mini').exists()).toBe(true)
})

test('should render active Tab', async () => {
  let testValue = ''
  const wrapper = mount({
    setup() {
      const activeName = ref('first')
      return () => (
        <Tab
          modelValue={activeName.value}
          onUpdate:modelValue={(value) => {
            activeName.value = value
            testValue = value
          }}
        >
          <TabPanel label="首页" name="first">
            1
          </TabPanel>
          <TabPanel label="用户管理" name="second">
            2
          </TabPanel>
          <TabPanel label="角色管理" name="third">
            3
          </TabPanel>
        </Tab>
      )
    },
  })
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)

  wrapper.find('.er-tabs-title a:nth-of-type(2)').trigger('click')
  await later()
  expect(testValue).toBe('second')
  wrapper.find('.er-tabs-title a:nth-of-type(3)').trigger('click')
  await later()
  expect(testValue).toBe('third')
  wrapper.find('.er-tabs-title a:nth-of-type(2)').trigger('click')
  await later()
  expect(testValue).toBe('second')
  wrapper.find('.er-tabs-title a:nth-of-type(1)').trigger('click')
  await later()
  expect(testValue).toBe('first')
})
