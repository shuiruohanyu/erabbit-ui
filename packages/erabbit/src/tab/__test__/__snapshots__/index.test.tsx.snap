// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

<<<<<<< HEAD
exports[`should render  mini size Tabs 1`] = `
"<div class=\\"er-tabs mini default top\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render  mini size Tab 1`] = `
"<div class=\\"er-tabs mini top\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render active Tabs 1`] = `
"<div class=\\"er-tabs default default top\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render active Tab 1`] = `
"<div class=\\"er-tabs default top\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render border-card type Tabs 1`] = `
=======
exports[`should render border-card type Tab 1`] = `
>>>>>>> upstream/main
"<div class=\\"er-tabs default border-card top\\">
  <div class=\\"er-tabs-title border-card\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container border-card\\">
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render bottom position Tabs 1`] = `
"<div class=\\"er-tabs default default bottom\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render bottom position Tab 1`] = `
"<div class=\\"er-tabs default bottom\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render bottom position and border-card type and mini size Tabs 1`] = `
=======
exports[`should render bottom position and border-card type and mini size Tab 1`] = `
>>>>>>> upstream/main
"<div class=\\"er-tabs mini border-card bottom\\">
  <div class=\\"er-tabs-title border-card\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container border-card\\">
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render bottom position and card type Tabs 1`] = `
=======
exports[`should render bottom position and card type Tab 1`] = `
>>>>>>> upstream/main
"<div class=\\"er-tabs default card bottom\\">
  <div class=\\"er-tabs-title card\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container card\\">
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render bottom position and mini size Tabs 1`] = `
"<div class=\\"er-tabs mini default bottom\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render bottom position and mini size Tab 1`] = `
"<div class=\\"er-tabs mini bottom\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render bottom position and small size Tabs 1`] = `
"<div class=\\"er-tabs small default bottom\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render bottom position and small size Tab 1`] = `
"<div class=\\"er-tabs small bottom\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render card type Tabs 1`] = `
=======
exports[`should render card type Tab 1`] = `
>>>>>>> upstream/main
"<div class=\\"er-tabs default card top\\">
  <div class=\\"er-tabs-title card\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container card\\">
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render default Tabs 1`] = `
"<div class=\\"er-tabs default default top\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render default Tab 1`] = `
"<div class=\\"er-tabs default top\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render left position Tabs 1`] = `
"<div class=\\"er-tabs default default left\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"height: 0px; transform: translateY(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render left position Tab 1`] = `
"<div class=\\"er-tabs default left\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"height: 0px; transform: translateY(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render right position Tabs 1`] = `
"<div class=\\"er-tabs default default right\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"height: 0px; transform: translateY(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render right position Tab 1`] = `
"<div class=\\"er-tabs default right\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"height: 0px; transform: translateY(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;

<<<<<<< HEAD
exports[`should render small size Tabs 1`] = `
"<div class=\\"er-tabs small default top\\">
  <div class=\\"er-tabs-title default\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container default\\">
=======
exports[`should render small size Tab 1`] = `
"<div class=\\"er-tabs small top\\">
  <div class=\\"er-tabs-title\\">
    <div class=\\"er-tabs-select-scroll\\" style=\\"width: 0px; transform: translateX(0px);\\"></div><a class=\\"active\\" href=\\"javascript:;\\">首页</a><a class=\\"\\" href=\\"javascript:;\\">用户管理</a><a class=\\"\\" href=\\"javascript:;\\">角色管理</a>
  </div>
  <div class=\\"er-tabs-container\\">
>>>>>>> upstream/main
    <div class=\\"er-tabs-panel current\\">1</div>
    <div class=\\"er-tabs-panel\\">2</div>
    <div class=\\"er-tabs-panel\\">3</div>
  </div>
</div>"
`;
